supertokens_python/supertokens.py

```diff
@@ -388,6 +398,131 @@ async def get_users(  # pylint: disable=no-self-use
 
         return UsersResponse(users, next_pagination_token)
 
+    async def create_user_id_mapping(  # pylint: disable=no-self-use
+        self,
+        supertokens_user_id: str,
+        external_user_id: str,
+        external_user_id_info: Optional[str] = None,
+        force: Optional[bool] = None,
+    ) -> Union[
+        CreateUserIdMappingOkResult,
+        UnknownSupertokensUserIDError,
+        UserIdMappingAlreadyExistsError,
+    ]:
+        querier = Querier.get_instance(None)
+
+        cdi_version = await querier.get_api_version()
+
+        if is_version_gte(cdi_version, "2.15"):
+            res = await querier.send_post_request(
+                NormalisedURLPath("/recipe/userid/map"),
+                {
+                    "supertokensUserId": supertokens_user_id,
+                    "externalUserId": external_user_id,
+                    "externalUserIdInfo": external_user_id_info,
+                    "force": force,
+                },
+            )
+            if res["status"] == "OK":
+                return CreateUserIdMappingOkResult()
+            if res["status"] == "UNKNOWN_SUPERTOKENS_USER_ID_ERROR":
+                return UnknownSupertokensUserIDError()
+            if res["status"] == "USER_ID_MAPPING_ALREADY_EXISTS_ERROR":
+                return UserIdMappingAlreadyExistsError(
+                    does_super_tokens_user_id_exist=res["doesSuperTokensUserIdExist"],
+                    does_external_user_id_exist=res["does_external_user_id_exist"],
+                )
+
+            raise_general_exception("Unknown response")
+
+        raise_general_exception("Please upgrade the SuperTokens core to >= 3.15.0")
+
+    async def get_user_id_mapping(  # pylint: disable=no-self-use
+        self,
+        user_id: str,
+        user_id_type: Optional[UserIDTypes] = None,
+    ) -> Union[GetUserIdMappingOkResult, UnknownMappingError]:
+        querier = Querier.get_instance(None)
+
+        cdi_version = await querier.get_api_version()
+
+        if is_version_gte(cdi_version, "2.15"):
+            res = await querier.send_get_request(
+                NormalisedURLPath("/recipe/userid/map"),
+                {
+                    "userId": user_id,
+                    "userIdType": user_id_type,
+                },
+            )
+            if res["status"] == "OK":
+                return GetUserIdMappingOkResult(
+                    supertokens_user_id=res["superTokensUserId"],
+                    external_user_id=res["externalUserId"],
+                    external_user_info=res["externalUserIdInfo"],
```

Comment:
this hasn't been tested properly cause the core is not guaranteed to return this.